src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs

```diff
@@ -185,7 +185,7 @@ private static bool IsError(Diagnostic diagnostic)
 
         public static Assembly LoadAssembly(MemoryStream assemblyStream, MemoryStream pdbStream)
         {
-            var assembly = AssemblyLoadContext.Default.LoadFromStream(assemblyStream, pdbStream);
+            var assembly = Assembly.Load(assemblyStream.GetBuffer(), pdbStream.GetBuffer());
```

Comment:
GetBuffer isn't right. That returns the entire underlying array even if it has more than what was written. You need to pass ToArray here.